File: frontend/src/app/sitemap.ts
/** sitemap.xml — dynamic route for Next.js App Router */

import type { MetadataRoute } from 'next'

const host = process.env.SITE_HOST ?? 'http://localhost:3000'

export const dynamic = 'force-static'

export default function sitemap(): MetadataRoute.Sitemap {
  const now = new Date()

  const routes: MetadataRoute.Sitemap = [
    {
      url: host,
      lastModified: now,
      changeFrequency: 'weekly',
      priority: 1,
    },
  ]

  return routes
}

File: docs/upgrade-v2/04-Web-Presence-SEO-GA4-i18n/src/config/sitemap.rb
# WEB-03 fix: host from env, regenerated on every deploy.
host ENV.fetch('SITE_HOST', 'www.vivim.net')

sitemap :site do
  url root_url, last_mod: Time.now, change_freq: 'daily', priority: 1.0
  url mobile_url, last_mod: Time.now, change_freq: 'weekly', priority: 0.7
  url why_ror_url, last_mod: Time.now, change_freq: 'weekly', priority: 0.7
end

ping_with "https://#{host}/sitemap.xml" if Rails.env.production?
